WordPress/gutenberg
# Mark applying block templates not persistent

## What?


The same issue appeared twice:

- [once](https://github.com/WordPress/gutenberg/pull/44596) when building a fallback for the navigation block
- [once](https://github.com/WordPress/gutenberg/pull/45776) when building a page list block that expands in list view

In both cases, we’re setting the template for a block asynchronously and that creates a persistent change in the post which seems to be because of [how templated blocks work](https://github.com/WordPress/gutenberg/blob/fecc1d2aa66c2fb2c6379fb47de5d7cd71f01f8c/packages/block-editor/src/components/inner-blocks/use-inner-block-template-sync.js#L84-L96)

This PR updates that to make sure applying templates does not create an undo level.

## Why?


Applying templates to block inner contents should not create undo levels because by being automattic it is decent to assume the content is already saved and it will not depend on the initial state of the template.


## How?



Using a simple `__unstableMarkNextChangeAsNotPersistent` call before applying the template.

## Testing Instructions

Easiest way to test is to checkout #45776 which implements this change.
Another way is to check out this PR and try to set block templates to custom blocks, core blocks, and also set the template async.

## Screenshots or screencast 

N/A
